src/cr/cube/stripe/cubemeasure.py

```diff
@@ -128,10 +128,20 @@ def unweighted_counts(self):
 class _BaseWeightedCubeCounts(_BaseCubeMeasure):
     """Base class for weighted-count cube-measure variants."""
 
+    def __init__(self, rows_dimension, weighted_counts):
+        super(_BaseWeightedCubeCounts, self).__init__(rows_dimension)
+        self._weighted_counts = weighted_counts
+
     @classmethod
     def factory(cls, cube, rows_dimension, ca_as_0th, slice_idx):
         """Return _BaseWeightedCubeCounts subclass instance appropriate to `cube`."""
-        raise NotImplementedError
+        if ca_as_0th:
```

Comment:
I'm open to a better name, but after walking the same path of confusion myself, I've ended up unable to come up with a better name. The payload from zz9 is interpreted differently when the first (0th) dimension is CA, and this is what this flag indicates.

I think what we _might_ be able to do is to encapsulate all the `ca_as_0th` rules inside `Cube` such that no one else needs to keep track of it. I think this would mean moving the responsibility for slicing the raw cube-measure payloads into Cube, which would be nice because that job is presently done very far away down in the cube-measures, several layers down, and that means that `ca_as_0th` has to make that entire trip, which is why it appears in so many calls.

I think that's actually a good direction to pursue next time we need to do some renovation in `cr.cube.cube.Cube`.